sklearn/decomposition/_pca.py

```diff
@@ -530,20 +538,20 @@ def _fit_full(self, X, n_components):
             )
 
         # Center data
-        self.mean_ = np.mean(X, axis=0)
+        self.mean_ = xp.mean(X, axis=0)
         X -= self.mean_
 
-        U, S, Vt = linalg.svd(X, full_matrices=False)
+        U, S, Vt = xp.linalg.svd(X, full_matrices=False)
```

Comment:
On `main`, this uses `scipy.linalg.svd`. This PR switches to using `numpy.linalg.svd` when container is `ndarray`.

Reply:
Is this a problem? Both use LAPACK from OpenBLAS in the end nowadays, no?

Reply:
I special cased the numpy case to continue calling scipy's linalg instead for now.